openbb_sdk/providers/alpha_vantage/openbb_alpha_vantage/models/stock_eod.py

```diff
@@ -0,0 +1,217 @@
+"""Alpha Vantage Stock End of Day fetcher."""
+
+
+from datetime import datetime
+from typing import Any, Dict, List, Literal, Optional, get_args
+
+import pandas as pd
+from dateutil.relativedelta import relativedelta
+from openbb_provider.abstract.fetcher import Fetcher
+from openbb_provider.standard_models.stock_eod import StockEODData, StockEODQueryParams
+from openbb_provider.utils.descriptions import DATA_DESCRIPTIONS
+from openbb_provider.utils.helpers import get_querystring
+from pydantic import Field, NonNegativeFloat, PositiveFloat, root_validator, validator
+
+
+class AlphaVantageStockEODQueryParams(StockEODQueryParams):
+    """Alpha Vantage Stock End of Day Query.
+
+    Source: https://www.alphavantage.co/documentation/
+    """
+
+    function: Literal[
```

Comment:
We should consider consolidating all interval choices into one, standardized parameter. Currently (Terminal and SDK V3), `monthly=True`.  Presently in SDK V4, there is:

- `interval="1mo"`
- `timespan="month"`
- `function="TIME_SERIES_MONTHLY"`
- `period="1mo"`

Reply:
@the-praxs I propose we keep the `function` attribute so we can still have the validators that check `available_on_functions` and `required_on_functions`. IMO, it also makes clear that not all functions require the same query args and them being required vary from function to function.

Something we can/should do is to hide the `function` field with `exclude=True` and set it up on a validator based on the `period`.

Wdyt

Reply:
I believe that having the `function` attribute reduces user experience because they need to provide the whole parameter. Also, this attribute shares the same look as the `period` which we can use to make it simpler for the end user, in line with the other providers. The necessary details can be mentioned in the descriptions.

If there is a way to make the period versatile and prevent erroneous parameter compatibility, I will be intrigued on how we can do this. I haven't tried the `exclude=True` option so if you can make it work it would be good.

Reply:
@the-praxs latest commit achieves this by having the function field as private. So the user does not have to worry about poor ux but we can keep validation for function specific fields!